opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +172,46 @@ public void testEventOnCreationAndDeletion() throws Exception {
         // delete the one with polling packages
         sendRequest(DELETE, "/monitoringLocations/location2", 204);
     }
+
+    @Test
+    @Transactional
+    public void testLocationLimits() throws Exception {
+
+        final Integer DEFAULT_LIMIT = 10;
+        final Integer LOCATION_COUNT = 15;
+
+        final ObjectMapper MAPPER = new ObjectMapper();
+        MAPPER.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        // There is one default location (therefore we subtract one from LOCATION COUNT)
+        for (int i = 0; i < LOCATION_COUNT - 1; i++) {
+            OnmsMonitoringLocation loc = new OnmsMonitoringLocation(String.format("LocationName-%05d",i),String.format("LocationArea-%05d", i));
+            sendData(POST, MediaType.APPLICATION_XML,"/monitoringLocations", JaxbUtils.marshal(loc), 201);
+        }
+
+        // null limit (empty parameters) should return default
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",
+                        Collections.emptyMap(),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(DEFAULT_LIMIT));
+
+        // limit less than default
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",Map.of("limit",
+                Integer.toString(DEFAULT_LIMIT - 1)),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(DEFAULT_LIMIT-1));
+
+        // limit equals default
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",Map.of("limit",
+                Integer.toString(DEFAULT_LIMIT)),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(DEFAULT_LIMIT));
+
+        // limit greater than default
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",Map.of("limit",
+                Integer.toString(DEFAULT_LIMIT + 1)),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(DEFAULT_LIMIT+1));
+
+        // max count
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",Map.of("limit",
+                Integer.toString(LOCATION_COUNT)),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(LOCATION_COUNT));
+
+        // unlimited, should return all
+        assertThat(MAPPER.readValue(sendRequest(GET,"/monitoringLocations",Map.of("limit",
+                Integer.toString(0)),200),OnmsMonitoringLocationDefinitionList.class).getCount(),is(LOCATION_COUNT));
```

Comment:
Please re-add the spaces to adhere minimal formatting rules and this will be fine.

Reply:
Reformatting done.